parachains/runtimes/bridge-hubs/bridge-hub-polkadot/src/xcm_config.rs

```diff
@@ -0,0 +1,265 @@
+		MultiLocation { parents: 1, interior: Here } |
+		MultiLocation { parents: 1, interior: X1(Plurality { id: BodyId::Executive, .. }) }
+	};
+	pub type ParentOrSiblings: impl Contains<MultiLocation> = {
+		MultiLocation { parents: 1, interior: Here } |
+		MultiLocation { parents: 1, interior: X1(_) }
+	};
+}
+/// A call filter for the XCM Transact instruction. This is a temporary measure until we properly
+/// account for proof size weights.
+///
+/// Calls that are allowed through this filter must:
+/// 1. Have a fixed weight;
+/// 2. Cannot lead to another call being made;
+/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
+pub struct SafeCallFilter;
+impl Contains<RuntimeCall> for SafeCallFilter {
+	fn contains(call: &RuntimeCall) -> bool {
+		#[cfg(feature = "runtime-benchmarks")]
+		{
+			if matches!(call, RuntimeCall::System(frame_system::Call::remark_with_event { .. })) {
+				return true
+			}
+		}
+
+		match call {
+			RuntimeCall::System(
+				frame_system::Call::set_heap_pages { .. } |
+				frame_system::Call::set_code { .. } |
+				frame_system::Call::set_code_without_checks { .. } |
+				frame_system::Call::kill_prefix { .. },
+			) |
+			RuntimeCall::ParachainSystem(..) |
+			RuntimeCall::Timestamp(..) |
+			RuntimeCall::Balances(..) |
+			RuntimeCall::CollatorSelection(
+				pallet_collator_selection::Call::set_desired_candidates { .. } |
+				pallet_collator_selection::Call::set_candidacy_bond { .. } |
+				pallet_collator_selection::Call::register_as_candidate { .. } |
+				pallet_collator_selection::Call::leave_intent { .. },
+			) |
+			RuntimeCall::Session(pallet_session::Call::purge_keys { .. }) |
+			RuntimeCall::XcmpQueue(..) |
+			RuntimeCall::DmpQueue(..) |
+			RuntimeCall::Utility(pallet_utility::Call::as_derivative { .. }) => true,
+			_ => false,
+		}
+	}
+}
+
+pub type Barrier = DenyThenTry<
+	DenyReserveTransferToRelayChain,
+	(
+		// Allow local users to buy weight credit.
+		TakeWeightCredit,
+		// Expected responses are OK.
+		AllowKnownQueryResponses<PolkadotXcm>,
+		WithComputedOrigin<
+			(
+				// Allow anything to pay for execution.
+				AllowTopLevelPaidExecutionFrom<Everything>,
+				// Parent and its exec plurality get free execution.
+				AllowExplicitUnpaidExecutionFrom<ParentOrParentsExecutivePlurality>,
+				// Subscriptions for version tracking are OK.
+				AllowSubscriptionsFrom<ParentOrSiblings>,
+			),
+			UniversalLocation,
+			ConstU32<8>,
+		>,
+	),
+>;
+
+pub struct XcmConfig;
+impl xcm_executor::Config for XcmConfig {
+	type RuntimeCall = RuntimeCall;
+	type XcmSender = XcmRouter;
+	type AssetTransactor = CurrencyTransactor;
+	type OriginConverter = XcmOriginToTransactDispatchOrigin;
+	// BridgeHub does not recognize a reserve location for any asset. Users must teleport DOT
+	// where allowed (e.g. with the Relay Chain).
+	type IsReserve = ();
+	/// Only allow teleportation of DOT.
+	type IsTeleporter = ConcreteNativeAssetFrom<DotRelayLocation>;
```

Comment:
We may need to revisit this as part of the end-to-end transfer story. Even though the Bridge Hub is part of the local consensus system for other paras under the same Relay, it acts as foreign consensus for many assets, so other chains (like Statemint) will want to reserve assets locally instead of trusting the BH with teleportation. Shouldn't prevent launch but we should make sure to consider this when we add any bridge pallets.

Reply:
ok, issue created here https://github.com/paritytech/parity-bridges-common/issues/1828